### PR TITLE
feature: Replace development logger with bunyan

### PR DESCRIPTION
**Pull request by lumberj**
Motivation:
To provide structured, machine-parseable logging for production
deployments.

The bunyan logger will by default log in JSON in one line:

```js
 {"name":"koa","hostname":"alan-lm","pid":39269,"req_id":"bbf85791-9169-43bd-a679-4214c8b9cea7","level":30,"req":{"method":"GET","url":"/accounts/test","headers":{"host":"localhost:3001","accept":"application/json","connection":"close"},"remoteAddress":"::ffff:127.0.0.1","remotePort":60024},"res":{"statusCode":200,"header":"HTTP/1.1 200 OK\r\nAccess-Control-Allow-Origin: *\r\nAccess-Control-Expose-Headers: link\r\nAccess-Control-Allow-Methods: GET,HEAD,PUT,POST,DELETE\r\nContent-Type: application/json; charset=utf-8\r\nContent-Length: 162\r\nDate: Tue, 12 Jul 2016 23:49:22 GMT\r\nConnection: close\r\n\r\n"},"duration":9,"msg":"  --> GET /accounts/test 200 9ms","time":"2016-07-12T23:49:22.811Z","v":0}
 ```

This can be parsed for human-readability using a JSON parser such as [`jq`](https://stedolan.github.io/jq/) or the [bunyan CLI](https://github.com/trentm/node-bunyan#cli-usage)

In this commit, `npm start` will by default pipe the JSON output to the
`bunyan` CLI. Example logging looks like:

```
 [2016-07-12T23:49:22.978Z]  INFO: koa/39269 on alan-lm:   <-- GET /connectors (req_id=0053ab9c-de7f-4782-a55f-a33ca6042f58, req.remoteAddress=::ffff:127.0.0.1, req.remotePort=60037)
    GET /connectors HTTP/1.1
    host: localhost:3001
    accept: application/json
    connection: close
[2016-07-12T23:49:22.982Z]  INFO: koa/39269 on alan-lm:   --> GET /connectors 200 4ms (req_id=0053ab9c-de7f-4782-a55f-a33ca6042f58, duration=4, req.remoteAddress=::ffff:127.0.0.1, req.remotePort=60037)
    GET /connectors HTTP/1.1
    host: localhost:3001
    accept: application/json
    connection: close
    --
    HTTP/1.1 200 OK
    HTTP/1.1 200 OK
...
```

Additionally, the request logger will log the `X-Request-ID` header and
store the result in the Koa context as `reqId` (e.g., `this.reqId`)
See:
https://github.com/koajs/bunyan-logger#koabunyanloggerrequestidcontextopts

### CODE REVIEW

File: src/services/log.js [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 
-const hub = require('mag-hub')
-const mag = require('mag')
-const log = require('five-bells-shared/lib/log')
+const bunyan = require('bunyan')
+const _ = require('lodash')
+const config = require('./config')
 
-module.exports = log(mag, hub)
+function createLogger (name) {
```
Comment (MatthewPhinney) [inline src/services/log.js]: Is this used?

**Comment by lumberj** [comment]
Unit  tests currently fail because it's not compatible with this, https://github.com/interledger/five-bells-shared/blob/master/testHelpers/log.js

Looking into the best solution

File: src/services/log.js (hunk 1 above)
Comment (lumberj) [inline src/services/log.js]: Yes, that's the function that is exported

File: package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "url": "git@github.com:interledger/five-bells-ledger.git"
   },
   "scripts": {
-    "start": "node src/app.js",
+    "start": "node src/app.js | bunyan",
```
Comment (MatthewPhinney) [inline package.json]: Does this pipe to stdout?

File: src/services/log.js (hunk 1 above)
Comment (MatthewPhinney) [inline src/services/log.js]: Oh yeah, I see now.

File: package.json (hunk 2 above)
Comment (lumberj) [inline package.json]: Yes

**Comment by lumberj** [comment]
It's not an obvious solution to get the unit tests passing with the new logger, so I'm going to close for now until I have the solution.

**Comment by MatthewPhinney** [comment]
Maybe we can double check with the five bells team, and then change the logger in `shared`.

**Comment by lumberj** [comment]
> Maybe we can double check with the five bells team, and then change the logger in shared.

The idea is to get it working in the ledger, then move common code to shared.

**Pull request closed without merging**